tests/phpunit/tests/post/getPostStatus.php

```diff
@@ -164,4 +241,88 @@ public function data_get_post_status_after_trashing() {
 			array( 'delete-attachment', 'publish', 'publish' ),
 		);
 	}
+
+	/**
+	 * Ensure the `get_post_states` function don't return the current filtered post status in its result array.
+	 *
+	 * @ticket 64026
+	 *
+	 * @dataProvider data_filtered_post_status_shouldnt_be_included_in_post_state_array
+	 *
+	 * @param string $post_state The post state to test.
+	 */
+	public function test_filtered_post_status_shouldnt_be_included_in_post_state_array( $post_state ) {
+		$_REQUEST['post_status'] = $post_state;
```

Comment:
Do we need there? We should restore it's value after the test otherwise it return unwanted status other unit tests.